api/src/__tests__/URL.test.js

```diff
@@ -0,0 +1,182 @@
+/* global describe, test, expect */
+
+const { Kind } = require('graphql/language')
+
+const { URL } = require('../types/URL')
+
+// this file taken and modified from https://github.com/okgrow/graphql-scalars
+
+/*
+ * these test cases are picked up from https://mathiasbynens.be/demo/url-regex
+ */
+
+const SHOULD_MATCH = [
+  'http://foo.com/blah_blah',
+  'http://foo.com/blah_blah/',
+  'http://foo.com/blah_blah_(wikipedia)',
+  'http://foo.com/blah_blah_(wikipedia)_(again)',
+  'http://www.example.com/wpstyle/?p=364',
+  'https://www.example.com/foo/?bar=baz&inga=42&quux',
+  'http://✪df.ws/123',
+  'http://userid:password@example.com:8080',
+  'http://userid:password@example.com:8080/',
+  'http://userid@example.com',
+  'http://userid@example.com/',
+  'http://userid@example.com:8080',
+  'http://userid@example.com:8080/',
+  'http://userid:password@example.com',
+  'http://userid:password@example.com/',
+  'http://142.42.1.1/',
+  'http://142.42.1.1:8080/',
+  'http://➡.ws/䨹',
+  'http://⌘.ws',
+  'http://⌘.ws/',
+  'http://foo.com/blah_(wikipedia)#cite-1',
+  'http://foo.com/blah_(wikipedia)_blah#cite-1',
+  'http://foo.com/unicode_(✪)_in_parens',
+  'http://foo.com/(something)?after=parens',
+  'http://☺.damowmow.com/',
+  'http://code.google.com/events/#&product=browser',
+  'http://j.mp',
+  'ftp://foo.bar/baz',
+  'http://foo.bar/?q=Test%20URL-encoded%20stuff',
+  'http://例子.测试',
+  'http://उदाहरण.परीक्षा',
+  'http://1337.net',
+  'http://a.b-c.de',
+  'http://223.255.255.254',
+  'http://localhost:8080',
+  'http://localhost:3000',
+  'http://127.0.0.1/',
+  'http://127.0.0.1:8080/',
+]
+
+const SHOULD_NOT_MATCH = [
+  'http://',
+  'http://.',
+  'http://..',
+  'http://../',
+  'http://?',
+  'http://??',
+  'http://??/',
+  'http://#',
+  'http://##',
+  'http://##/',
+  'http://foo.bar?q=Spaces should be encoded',
+  '//',
+  '//a',
+  '///a',
+  '///',
+  'http:///a',
+  'foo.com',
+  'rdar://1234',
+  'h://test',
+  'http:// shouldfail.com',
+  'http://foo.bar/foo(bar)baz quux',
+  'ftps://foo.bar/',
+  'http://-error-.invalid/',
+  // 'http://a.b--c.de/',
+  'http://-a.b.co',
+  'http://a.b-.co',
+  'http://0.0.0.0',
+  'http://10.1.1.0',
+  'http://10.1.1.255',
+  'http://224.1.1.1',
+  'http://1.1.1.1.1',
+  'http://123.123.123',
+  'http://3628126748',
+  'http://.www.foo.bar/',
+  // 'http://www.foo.bar./',
```

Comment:
🔥

Reply:
Yeah, I've got to take a closer look at the tests, just wanted to get something started before leaving yesterday. There are a couple not passing.

Reply:
Oh, I was referring to deleting the commented out line (or uncomment) :)

Reply:
ahhhhh...ok! thanks Steve